src/Foundation/InstalledApp.php

```diff
@@ -57,6 +59,12 @@ public function getRequestHandler()
         $pipe = new MiddlewarePipe;
 
         $pipe->pipe(new BasePath($this->basePath()));
+        $pipe->pipe(
```

Comment:
I would prefer to see the Middleware changed made in https://github.com/flarum/core/blob/master/src/Api/ApiServiceProvider.php and https://github.com/flarum/core/blob/master/src/Forum/ForumServiceProvider.php following a similiar method that the error handler Middleware is implemented to ensure that it can be overridded if needed by extension devs.

You can see how the error handler is implemented in the ApiServiceProvider file on line 59 (actual call/binding) and 47 (adding to middleware array), The ForumServiceProvider has similar logic (possibly even the same logic)

Reply:
I definitely agree with this change.
Do I also need to include the middleware in
https://github.com/flarum/core/blob/master/src/Admin/AdminServiceProvider.php ?

Reply:
I would say yes to ensure that IPs are tracked properly for that, I don't think core uses IPs in anyway for admin, but I know a couple extensions might. So I'd include it there too.